sequencer/src/genesis.rs

```diff
@@ -74,6 +76,25 @@ impl Genesis {
     }
 }
 
+impl Genesis {
+    pub async fn validate_fee_contract(&self, l1_rpc_url: String) -> anyhow::Result<()> {
+        let provider = Provider::<Http>::try_from(l1_rpc_url)?;
+
+        if let Some(fee_contract_address) = self.chain_config.fee_contract {
+            if !is_proxy_contract(provider, fee_contract_address)
+                .await
+                .expect("Failed to determine if fee contract is a proxy")
+            {
+                panic!("Fee contract's address is not a proxy");
+            }
+        } else {
+            panic!("Fee contract's address is missing from the chain config!");
+        }
+
```

Comment:
Also would be great if we can add the check for upgrades. So, we iterate over each upgrade -> get the chain config -> validate the contract address

Reply:
ah gotcha, done here with supporting tests https://github.com/EspressoSystems/espresso-sequencer/pull/2000